### ISSUE TITLE
[PLUGIN REQUEST] harpoon2

### ISSUE THREAD

**Issue by nicolasauler** [nix-community/nixvim#1440]
| Field    | Description |
|----------|-------------|
| Plugin   | `<PLUGIN>`  | harpoon2
| Homepage | `<URL>`     | https://github.com/ThePrimeagen/harpoon/tree/harpoon2
| Nixpkgs  | `<PACKAGE>` | https://search.nixos.org/packages?channel=unstable&show=vimPlugins.harpoon2&from=0&size=50&sort=relevance&type=packages&query=harpoon2

**Comment by 4lxs** [comment]
i am pretty sure this has been discussed and decided that we're waiting for harpoon2 to become the default before switching.

**Comment by GaetanLepage** [comment]
Indeed. [Harpoon 2 has not yet been officially released](https://github.com/ThePrimeagen/harpoon/discussions/550). Hence, in the meantime, we will sick on harpoon 1. I have already started to adapt the nixvim module on my side.

I will leave this issue open to track this migration.

**Comment by nicolasauler** [comment]
> i am pretty sure this has been discussed and decided that we're waiting for harpoon2 to become the default before switching.

Sorry mate!
I checked the issues for anything harpoon2 related, but couldn't find any open or closed ones, that's why I opened this one.


> I will leave this issue open to track this migration.

Great! Thanks Gaetan.

**Comment by GaetanLepage** [comment]
> Sorry mate!
> I checked the issues for anything harpoon2 related, but couldn't find any open or closed ones, that's why I opened this one.

No problem :) Maybe it was on the matrix or somewhere else, I don't remember.

**Comment by MattSturgeon** [comment]
As per discussion on matrix, I think we should add harpoon2 alongside harpoon.

- Rename `plugins.harpoon` to `plugins.harpoon1`
- Add `pluigins.harpoon2`
- Add a "[transition alias](https://github.com/nix-community/nixvim/pull/1657)" pointing to either `plugins.harpoon1` or `plugins.harpoon2`
- Print a warning when the alias is used, such as:
  _"`plugins.harpoon` is an alias for `plugins.harpoon1`. After 24.11 `plugins.harpoon2` will be moved to `plugins.harpoon`. Use `plugins.harpoon1` or `plugins.harpoon2` to silence this warning"_

One thing I _don't_ like about this approach, is it implies we will always have three `plugins.harpoon*` modules (one of which being an alias), though idk if that's a real issue or just a small ugliness?
